htmresearch/algorithms/column_pooler.py

```diff
@@ -296,34 +302,44 @@ def _chooseCells(self, candidates, n, numActiveSegmentsByCell):
     @return (list) Cells to activate
     """
 
-    orderedCandidates = sorted(candidates,
-                               key=numActiveSegmentsByCell.__getitem__,
-                               reverse=True)
-    activeSegmentCounts = sorted(set(numActiveSegmentsByCell[cell]
-                                     for cell in candidates),
-                                 reverse=True)
+    if n <= 0:
+      return numpy.empty(0)
+
+    # Keep a list of numpy arrays.
+    allChosenCells = []
+    numChosen = 0
 
-    chosenCells = []
-    i = 0
+    # Walk the active segment counts in decreasing order.
+    activeSegmentCounts = numpy.unique(
```

Comment:
It actually creates a "view" on a numpy array. It doesn't copy the array.

`numActiveSegmentsByCell[candidates]` creates an array.
`np.unique` creates an array.
`[::-1]` creates a view.

So yes, you could say that this is effectively an iterator.